frontend/src/pages/modelRegistry/screens/ModelPropertiesTableRow.tsx

```diff
@@ -0,0 +1,176 @@
+import * as React from 'react';
+import { ActionsColumn, Td, Tr } from '@patternfly/react-table';
+import {
+  ActionList,
+  ActionListItem,
+  Button,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { CheckIcon, TimesIcon } from '@patternfly/react-icons';
+import { KeyValuePair } from '~/types';
+import { EitherNotBoth } from '~/typeHelpers';
+
+type ModelPropertiesTableRowProps = {
+  allExistingKeys: string[];
+  setIsEditing: (isEditing: boolean) => void;
+  isSavingEdits: boolean;
+  setIsSavingEdits: (isSaving: boolean) => void;
+  saveEditedProperty: (oldKey: string, newPair: KeyValuePair) => Promise<unknown>;
+} & EitherNotBoth<
+  { isAddRow: true },
+  {
+    isEditing: boolean;
+    keyValuePair: KeyValuePair;
+    deleteProperty: (key: string) => Promise<unknown>;
+  }
+>;
+
+const ModelPropertiesTableRow: React.FC<ModelPropertiesTableRowProps> = ({
+  isAddRow,
+  isEditing = isAddRow,
+  keyValuePair = { key: '', value: '' },
+  deleteProperty = () => Promise.resolve(),
+  allExistingKeys,
+  setIsEditing,
+  isSavingEdits,
+  setIsSavingEdits,
+  saveEditedProperty,
+}) => {
+  const { key, value } = keyValuePair;
+  const [unsavedKey, setUnsavedKey] = React.useState(key);
+  const [unsavedValue, setUnsavedValue] = React.useState(value);
+
+  const keyValidationError =
+    unsavedKey !== key && allExistingKeys.includes(unsavedKey)
+      ? 'Key must not match an existing property key or label'
+      : unsavedKey.length > 63
```

Comment:
nit.. wonder if it would be better to just use normal if/else blocks here just for readability

Reply:
Good call